client/src/JavaProjectTypes.ts

```diff
@@ -0,0 +1,60 @@
+export enum JavaType {
+  Interface = "Interface",
+  Enum = "Enum",
+  Abstract = "Abstract",
+  Concrete = "Concrete",
+}
+
+export enum JavaAccessModifier {
+  Private = "private",
+  Public = "public",
+  Protected = "protected",
+}
+
+export type JavaField = {
+  name: string;
+  type: string;
+  accessModifier: JavaAccessModifier;
+};
+
+export type JavaFieldsMap = {
+  [key: string]: JavaField[];
+};
+
+export type JavaParametersMap = {
+  [key: string]: string;
+};
+
+export type JavaMethod =
+  | {
+      name: string;
+      accessModifier: JavaAccessModifier;
+      parameters: JavaParametersMap;
+      returnType: string;
+    }
+  | {
+      name: string;
+      accessModifier: JavaAccessModifier;
+      parameters: JavaParametersMap;
+      returnType: null; //for constructor
```

Comment:
I think `returnType` should be removed as a constructor returns nothing, not even `null`. ` I don't think it is accurate to classify a constructor as a method. We should create a separate type `JavaConstructor` for it.

Reply:
I just followed the readme. should I change the readme as well?

Reply:
I've updated README in #26